Lib/fontTools/cffLib/cff2mergePen.py

```diff
@@ -0,0 +1,303 @@
+# Copyright (c) 2009 Type Supply LLC
+
+from __future__ import print_function, division, absolute_import
+from fontTools.misc.fixedTools import otRound
+from fontTools.misc.psCharStrings import CFF2Subr
+from fontTools.pens.t2CharStringPen import T2CharStringPen
+from fontTools.cffLib.specializer import (commandsToProgram,
+										specializeCommands,
+										)
+
+
+class MergeTypeError(TypeError):
+	def __init__(self, point_type, pt_index, m_index, default_type):
+		self.error_msg = [
+					"'{point_type}' at point index {pt_index} in master "
+					"index {m_index} differs from the default font point "
+					"type '{default_type}'".format(
+						point_type=point_type,
+						pt_index=pt_index,
+						m_index=m_index,
+						default_type=default_type)
+					][0]
+		super(MergeTypeError, self).__init__(self.error_msg)
+
+
+class CFF2CharStringMergePen(T2CharStringPen):
+	"""Pen to merge Type 2 CharStrings.
+	"""
+	def __init__(self, default_commands,
+				 num_masters, master_idx, roundTolerance=0.5):
+		super(
+			CFF2CharStringMergePen,
+			self).__init__(width=None,
+						   glyphSet=None, CFF2=True,
+						   roundTolerance=roundTolerance)
+		self.pt_index = 0
+		self._commands = default_commands
+		self.m_index = master_idx
+		self.num_masters = num_masters
+		self.prev_move_idx = 0
+		self.roundTolerance = roundTolerance
+
+	def _round(self, number):
+		tolerance = self.roundTolerance
+		if tolerance == 0:
+			return number  # no-op
+		rounded = otRound(number)
+		# return rounded integer if the tolerance >= 0.5, or if the absolute
+		# difference between the original float and the rounded integer is
+		# within the tolerance
+		if tolerance >= .5 or abs(rounded - number) <= tolerance:
+			return rounded
+		else:
+			# else return the value un-rounded
+			return number
+
+	def _p(self, pt):
+		""" Unlike T2CharstringPen, this class stores absolute values.
+		This is to allow the logic in check_and_fix_clospath() to work,
+		where the current or previous absolute point has to be compared to
+		the path start-point.
+		"""
+		self._p0 = pt
+		return list(self._p0)
+
+	def make_flat_curve(self, prev_coords, cur_coords):
+		# Convert line coords to curve coords.
+		dx = self._round((cur_coords[0] - prev_coords[0])/3.0)
+		dy = self._round((cur_coords[1] - prev_coords[1])/3.0)
+		new_coords = [prev_coords[0] + dx,
+					  prev_coords[1] + dy,
+					  prev_coords[0] + 2*dx,
+					  prev_coords[1] + 2*dy
+					  ] + cur_coords
+		return new_coords
+
+	def make_curve_coords(self, coords, is_default):
+		# Convert line coords to curve coords.
+		prev_cmd = self._commands[self.pt_index-1]
+		if is_default:
+			new_coords = []
+			for i, cur_coords in enumerate(coords):
+				prev_coords = prev_cmd[1][i]
+				master_coords = self.make_flat_curve(
+											prev_coords[:2], cur_coords
```

Comment:
unnecessary space

Reply:
The FontTools codebase does not abide to PEP 8, but when adding new files I think we should make the effort of getting as close to that directive as possible.

Reply:
> What is the current policy in fonttools on tab indentation?

we don't have any. I prefer we use spaces for new code, though most of the old modules in fonttools use tabs.
I would rather run an autoformatter like [black](https://github.com/ambv/black) on the whole codebase and spare ourselves all this. I don't care how the code is formatted as long as it's consistent and good enough, and bots can take care the chore while I focus on the content.
Other developers like to write poems and may disagree.